src/main/java/com/hcl/appscan/jenkins/plugin/scanners/DynamicAnalyzer.java

```diff
@@ -129,46 +138,117 @@ public void setOptimization(String optimization) {
+		if (RECORDED.equals(m_loginType))
+			m_trafficFile = trafficFile;
+	}
+
+	public String getTrafficFile() {
+		return m_trafficFile;
+	}
+
 	@Override
 	public String getType() {
 		return DYNAMIC_ANALYZER;
 	}
-	
+
+	public String isLoginTypes(String loginTypeName) {
+		if (m_loginType != null) {
+			return m_loginType.equalsIgnoreCase(loginTypeName) ? "true" : "";
+		} else if((m_loginUser != null || m_loginPassword !=null)){
+			m_loginType = AUTOMATIC;
+			return "true";
+		} else if (loginTypeName.equals(NONE)) { //Default
+			return "true";
+		} else {
+			return "";
+		}
+	}
+
+	public String upgradeLoginScenario(){
+		if(m_loginUser != null || m_loginPassword != null){
+			return m_loginType = AUTOMATIC;
+		} else {
+			return m_loginType = NONE;
+		}
+	}
+
 	@Override
-	public Map<String, String> getProperties(VariableResolver<String> resolver) {
+	public Map<String, String> getProperties(VariableResolver<String> resolver) throws hudson.AbortException {
 		Map<String, String> properties = new HashMap<String, String>();
-                if(resolver == null) {
+		if (resolver == null) {
 			properties.put(TARGET, getTarget());
-			properties.put(LOGIN_USER, m_loginUser);
-			properties.put(LOGIN_PASSWORD, Secret.toString(m_loginPassword));
 			properties.put(SCAN_FILE, m_scanFile);
 			properties.put(EXTRA_FIELD, m_extraField);
-                }
-                else {
+			if(m_loginType == null){
+				m_loginType = upgradeLoginScenario();
+			}
+				if (Objects.equals(m_loginType, RECORDED)) {
+					properties.put(TRAFFIC_FILE, m_trafficFile);
+					if (m_trafficFile.equals("")) {
+						throw new hudson.AbortException(Messages.error_login_fields_empty_manual());
+					} else if ((!(new File(m_trafficFile).isFile()))){
+						throw new hudson.AbortException(Messages.error_login_fields_manual());
+					}
+				} else if (Objects.equals(m_loginType, AUTOMATIC)) {
+					properties.put(LOGIN_USER, m_loginUser);
+					properties.put(LOGIN_PASSWORD, Secret.toString(m_loginPassword));
+					if(m_loginUser.equals("") || m_loginPassword.equals(Secret.fromString(""))){
+						throw new hudson.AbortException(Messages.error_login_fields_automatic());
+					}
+				}
+		} else {
 			properties.put(TARGET, Util.replaceMacro(getTarget(), resolver));
-			properties.put(LOGIN_USER, Util.replaceMacro(m_loginUser, resolver));
-			properties.put(LOGIN_PASSWORD, Util.replaceMacro(Secret.toString(m_loginPassword), resolver));
 			properties.put(SCAN_FILE, resolvePath(m_scanFile, resolver));
 			properties.put(EXTRA_FIELD, Util.replaceMacro(m_extraField, resolver));
-                }
-                properties.put(SCAN_TYPE, m_scanType);
-                properties.put(TEST_OPTIMIZATION_LEVEL, m_optimization);
-                properties.put(PRESENCE_ID, m_presenceId);
+			if(m_loginType == null || m_loginType.equals("")){
+				m_loginType = upgradeLoginScenario();
+			}
+			if (Objects.equals(m_loginType, RECORDED)) {
```

Comment:
Why is `Objects.equals()` being used here instead of just comparing the string values with `RECORDED.equals(m_loginType)`?